PhysicsTools/NanoAOD/plugins/GenWeightsTableProducer.cc

```diff
@@ -276,6 +277,25 @@ class GenWeightsTableProducer : public edm::global::EDProducer<edm::StreamCache<
                                         break;
                                     }
                                 }
+                            } else if (groups.str(2) == "NNPDF30_lo_as_0130.LHgrid") {
+                                for ( ++iLine; iLine < nLines; ++iLine) {
+                                    if (lheDebug) std::cout << "    " << lines[iLine];
+                                    if (std::regex_search(lines[iLine], groups, pdfwOld)) {
+                                        unsigned int lhaID = std::stoi(groups.str(2))+262000;
```

Comment:
hi @arizzi - what is 262000? would you add a comment? [or for that matter why "NNPDF30..."?

Reply:
262000 is the LHA ID of NNPDF30_lo_as_0118.
For reasons unknown to me, while MC samples have PDFs weights stored in a logical way (the header is a generic pdf weight header, and the weights IDs contain the LHA ID of the PDF), some 80X MCs have PDF weights coded in a different way, with the PDF name hard-coded in the header, and the LHA ID is not written anywhere.
I can add a comment.

Reply:
Thanks!